src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2689,6 +2762,33 @@ def update_node_restriction(self, mc: ManagedCluster) -> ManagedCluster:
 
         return mc
 
+    def update_vpa(self, mc: ManagedCluster) -> ManagedCluster:
+        """Update workload auto-scaler profile vertical pod auto-scaler for the ManagedCluster object.
+        
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        if self.context.get_enable_vpa():
+            if mc.workload_auto_scaler_profile is None:
+                mc.workload_auto_scaler_profile = self.models.ManagedClusterWorkloadAutoScalerProfile
```

Comment:
```suggestion
                mc.workload_auto_scaler_profile = self.models.ManagedClusterWorkloadAutoScalerProfile()
```

Reply:
Thanks for the comment. Updated in the latest iteration.